server/cache_util/cachefactory.py

```diff
@@ -55,6 +55,8 @@ def pickAvailableCache(sizeEach, portion=8):
     else:
         memory = 1024 ** 3
     numItems = max(int(math.floor(memory / portion / sizeEach)), 2)
+    if maxItems:
+        numItems = max(numItems, maxItems)
```

Comment:
Shouldn't this be `min(numItems, maxItems)`? Also we should add a docstring for the new param.